include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+    """
+    Signals that a buffer is not big enough.
+    """
+
+    pass
+
+
+class OutOfRangeError(Exception):
+    """
+    Signals that an index-like parameter has a value that is outside its
+    possible values.
+    """
+
+    pass
+
+
+class LogicErrorError(Exception):
+    """
+    Signals faulty logic within the program such as violating logical
+    preconditions or class invariants and may be preventable.
+    """
+
+    pass
+
+
+class DeletedObjectError(Exception):
+    """
+    Signals accessing an already deleted object.
+    """
+
+    pass
+
+
+class InvalidArgumentError(Exception):
+    """
+    Signals that some of the arguments have invalid values.
+    """
+
+    pass
+
+
+class KeyAlreadyExistsError(Exception):
+    """
+    Signals that a key already exists in a container-like object.
+    """
+
+    pass
+
+
+class ImmutableObjectError(Exception):
+    """
+    Signals modification of an immutable object.
+    """
+
+    pass
+
+
+class ValueConversionError(Exception):
+    """
+    Signals that conversion between python and cypher values failed.
+    """
+
+    pass
+
+
+class Label:
+    __slots__ = ("_name",)
+
+    def __init__(self, name: str):
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, Label):
+            return self._name == other.name
+
+        if isinstance(other, str):
+            return self._name == other
+
+        return NotImplemented
+
+
+# Named property value of a Vertex or an Edge.
+# It would be better to use typing.NamedTuple with typed fields, but that is
+# not available in Python 3.5.
+Property = namedtuple("Property", ("name", "value"))
+
+
+class Properties:
+    __slots__ = ("_vertex_or_edge", "_len")
+
+    def __init__(self, vertex_or_edge):
+        if not isinstance(vertex_or_edge, (_mgp_mock.Vertex, _mgp_mock.Edge)):
+            raise TypeError(f"Expected _mgp_mock.Vertex or _mgp_mock.Edge, got {type(vertex_or_edge)}")
+
+        self._len = None
+        self._vertex_or_edge = vertex_or_edge
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, as the underlying C API should
+        # not support deepcopy. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Edge and _mgp_mock.Vertex types as they are actually references
+        # to graph elements and not proper values.
+        return Properties(self._vertex_or_edge)
+
+    def get(self, property_name: str, default=None) -> object:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        try:
+            return self[property_name]
+        except KeyError:
+            return default
+
+    def set(self, property_name: str, value: object) -> None:
+        if nx.is_frozen(self._vertex_or_edge.graph):
+            raise ImmutableObjectError("Cannot modify immutable object.")
```

Comment:
Can you push this part to `_mgp_mock.Vertex` so you have following functionality:
```
if self._vertex_or_edge.is_frozen():
    raise ... 
```

I think `nx` should be hidden in `_mgp_mock` as it is encapsulation for that part, right? Does that make sense?

Reply:
Good idea! I initially created `_mgp_mock` only so that `mgp_mock` classes have the exact same methods as `mgp`, but this makes more sense from an engineering perspective.

Reply:
Done